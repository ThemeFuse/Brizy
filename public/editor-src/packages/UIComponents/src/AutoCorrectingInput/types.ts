import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
