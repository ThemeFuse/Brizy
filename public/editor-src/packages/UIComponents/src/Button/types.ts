import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
