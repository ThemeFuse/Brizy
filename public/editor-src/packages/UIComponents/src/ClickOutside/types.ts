import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
