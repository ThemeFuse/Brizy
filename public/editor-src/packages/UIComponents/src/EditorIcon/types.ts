export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
