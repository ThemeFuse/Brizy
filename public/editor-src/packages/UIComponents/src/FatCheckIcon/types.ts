import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
