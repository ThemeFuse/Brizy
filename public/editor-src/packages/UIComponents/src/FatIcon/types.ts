import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
