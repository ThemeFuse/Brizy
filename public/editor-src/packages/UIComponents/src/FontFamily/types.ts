import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
