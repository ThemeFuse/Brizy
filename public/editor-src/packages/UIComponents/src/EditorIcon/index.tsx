import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
