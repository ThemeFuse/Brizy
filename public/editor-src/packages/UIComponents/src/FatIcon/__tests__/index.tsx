import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import _ from "underscore";
import { IconsName } from "../../EditorIcon/types";
import { FatIcon } from "../index";

describe("Test FatIcon", () => {
  const onClick = jest.fn();
  const iconName = IconsName["t2-event-calendar"];

  describe("Snapshots", () => {
    test("FatIcon snapshot", () => {
      const icon = render(
        <FatIcon onClick={onClick} icon={iconName} label={"Label"} />
      );

      expect(icon.container.firstChild).toMatchSnapshot();
    });

    test("FatIcon active", async () => {
      const { container } = render(
        <FatIcon onClick={onClick} icon={iconName} label={"Label"} isActive />
      );

      await waitFor(() => {
        const fatIcon = container.getElementsByClassName("brz-ed--fat-icon")[0];
        const wrapper = container.getElementsByClassName(
          "brz-ed--fat-icon__wrapper"
        )[0];
        const icon = container.getElementsByClassName(
          "brz-ed--fat-icon-icon"
        )[0];
        const label = container.getElementsByClassName(
          "brz-ed--fat-icon__label"
        )[0];

        expect(fatIcon).toHaveClass("brz-ed--fat-icon__active");
        expect(wrapper).toHaveClass("brz-ed--fat-icon__wrapper--active");
        expect(icon).toHaveClass("brz-ed--fat-icon-icon--active");
        expect(label).toHaveClass("brz-ed--fat-icon__label--active");

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });

  test("Check click event", () => {
    const { container } = render(
      <FatIcon onClick={onClick} icon={iconName} label={"Label"} />
    );
    const nodes = container.querySelectorAll("div");
    const icon = nodes[nodes.length - 1];

    icon && fireEvent.click(icon);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("renders the label prop correctly", () => {
    const label = "Test Label";
    const { getByText } = render(
      <FatIcon
        isActive={false}
        className="test-class"
        icon={iconName}
        label={label}
        onClick={_.noop}
      />
    );
    expect(getByText(label)).toBeInTheDocument();
  });
});
