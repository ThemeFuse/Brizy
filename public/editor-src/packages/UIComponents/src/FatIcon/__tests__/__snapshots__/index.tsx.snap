// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FatIcon Snapshots FatIcon active 1`] = `
<div
  class="brz-ed--fat-icon brz-ed--fat-icon__active"
>
  <div
    class="brz-ed--fat-icon__wrapper brz-ed--fat-icon__wrapper--active"
  >
    <svg
      class="brz-icon-svg brz-ed-icon-svg align-[initial] brz-ed--fat-icon-icon brz-ed--fat-icon-icon--active"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <g
        fill="currentColor"
        fill-rule="evenodd"
        stroke="none"
        stroke-width="1"
      >
        <path
          d="M16,3.5 L16,14.5 C16,15.3284505 15.2325149,16 14.2857143,16 L1.71428571,16 C0.767485119,16 0,15.3284505 0,14.5 L0,3.5 C0,2.67154948 0.767485119,2 1.71428571,2 L3.42857143,2 L3.42857143,0.375 C3.42857143,0.16796875 3.62053571,0 3.85714286,0 L5.28571429,0 C5.52232143,0 5.71428571,0.16796875 5.71428571,0.375 L5.71428571,2 L10.2857143,2 L10.2857143,0.375 C10.2857143,0.16796875 10.4776786,0 10.7142857,0 L12.1428571,0 C12.3794643,0 12.5714286,0.16796875 12.5714286,0.375 L12.5714286,2 L14.2857143,2 C15.2325149,2 16,2.67154948 16,3.5 Z M14,13.8223684 L14,5 L2,5 L2,13.8223684 C2,13.9204359 2.09161932,14 2.20454545,14 L13.7954545,14 C13.9083807,14 14,13.9204359 14,13.8223684 Z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </g>
    </svg>
  </div>
  <div
    class="brz-ed--fat-icon__label brz-ed--fat-icon__label--active"
  >
    Label
  </div>
</div>
`;

exports[`Test FatIcon Snapshots FatIcon snapshot 1`] = `
<div
  class="brz-ed--fat-icon"
>
  <div
    class="brz-ed--fat-icon__wrapper"
  />
  <div
    class="brz-ed--fat-icon__label"
  >
    Label
  </div>
</div>
`;
